pkg/controller/route/routecontroller.go

```diff
@@ -162,60 +163,36 @@ func (rc *RouteController) reconcile(nodes []api.Node, routes []*cloudprovider.R
 	return nil
 }
 
-func updateNetworkingCondition(node *api.Node, routeCreated bool) {
-	_, networkingCondition := api.GetNodeCondition(&node.Status, api.NodeNetworkUnavailable)
-	currentTime := unversioned.Now()
-	if routeCreated {
-		if networkingCondition != nil && networkingCondition.Status != api.ConditionFalse {
-			networkingCondition.Status = api.ConditionFalse
-			networkingCondition.Reason = "RouteCreated"
-			networkingCondition.Message = "RouteController created a route"
-			networkingCondition.LastTransitionTime = currentTime
-		} else if networkingCondition == nil {
-			node.Status.Conditions = append(node.Status.Conditions, api.NodeCondition{
+func (rc *RouteController) updateNetworkingCondition(nodeName string, routeCreated bool) error {
+	var err error
+	for i := 0; i < updateNodeStatusMaxRetries; i++ {
+		// Patch could also fail, even though the chance is very slim. So we still do
```

Comment:
Could you check the error type and only retry on conflict? (using pkg/api/errors.IsConflict)